docs/wrapper-layer.md

```diff
@@ -0,0 +1,53 @@
+The OmnICU Wrapper Layer
+========================
+
+A key piece of rolling out ergonomic OmnICU APIs for Rust and other target languages is the *wrapper layer*: code that sits between the client and the lower-level OmnICU libraries.  This document explains the design behind the OmnICU wrapper layer.
+
+## Background
+
+### Ergonomic API versus Logical API
+
+When a programmer writes code using Intl, the code they write should demonstrate their *intent*.  They should use what I'm calling the *ergonomic API* of OmnICU.
+
+A *logical API*, on the other hand, is one that is closely tied to specific to functional units, like currencies or measurement units.  A logical API function should require a specific hunk of data and perform a specific task.
+
+Although ergonomic APIs correspond with what users of i18n intend to do, from an implementation standpoint, they are often quite heavy-weight, pulling in many dependencies that may not be required for all call sites.  Intl.NumberFormat is an example: it supports currency formatting, which requires a lot of code and data, but when formatting a number that's not a currency, you shouldn't need to carry all that extra code and data.
+
+The goal of the *wrapper layer* is to provide an ergonomic API for clients, but map it to the logical API under the hood.
```

Comment:
So to clarify:
1. Wrapper layer implements ergonomic API
2. Is resolving user request (options, locale,...) into potential data fetch, and is passing data to logical API under the hood.